src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -516,26 +511,49 @@ public String createServiceBroker(CloudServiceBroker serviceBroker) {
     @Override
     public CloudServiceKey createAndFetchServiceKey(CloudServiceKey keyModel, String serviceInstanceName) {
```

Comment:
The whole idea of this method is quite bad. The method is not idempotent and in case of retry, it will fail.

Reply:
I agree, this is bad. I will track a separate backlog for this because it will need a complete refactoring of the parallelKeyHandler (gacd), and should be considered what to use, the flowable's polling or a complete new java polling